backend/danswer/connectors/teams/connector.py

```diff
@@ -0,0 +1,269 @@
+            filter_str = f"last_modified_datetime ge {start.isoformat()} and last_modified_datetime le {end.isoformat()}"
+
+        channel_list: list[Channel] = []
+        for team_object in team_object_list:
+            query = team_object.channels.get()
+            if filter_str:
+                query = query.filter(filter_str)
+            channel_objects = query.execute_query()
+            channel_list.extend(channel_objects)
+
+        return channel_list
+
+    def get_all_team_objects(self) -> list[Team]:
+        if self.graph_client is None:
+            raise ConnectorMissingCredentialError("Teams")
+
+        team_object_list: list[Team] = []
+
+        teams_object = self.graph_client.teams.get().execute_query()
+
+        if len(self.requested_team_list) > 0:
+            for requested_team in self.requested_team_list:
+                adjusted_request_string = requested_team.replace(" ", "")
+                for team_object in teams_object:
+                    adjusted_team_string = team_object.display_name.replace(" ", "")
+                    if adjusted_team_string == adjusted_request_string:
+                        team_object_list.append(team_object)
+        else:
+            team_object_list.extend(teams_object)
+
+        return team_object_list
+
+    def _fetch_from_teams(
+        self, start: datetime | None = None, end: datetime | None = None
+    ) -> GenerateDocumentsOutput:
+        if self.graph_client is None:
+            raise ConnectorMissingCredentialError("Teams")
+
+        team_object_list = self.get_all_team_objects()
+
+        channel_list = self.get_channel_object_list_from_team_list(
+            team_object_list=team_object_list,
+            start=start,
+            end=end,
+        )
+
+        # goes over channels, converts them into Document objects and then yields them in batches
+        doc_batch: list[Document] = []
+        batch_count = 0
+        for channel_object in channel_list:
+            post_message_lists = self.get_post_message_lists_from_channel(
+                channel_object
+            )
+            for base_message_groups in post_message_lists:
+                doc_batch.append(
+                    self.convert_post_message_list_to_document(
+                        channel_object, base_message_groups
+                    )
+                )
+
+            batch_count += 1
+            if batch_count >= self.batch_size:
+                yield doc_batch
+                batch_count = 0
+                doc_batch = []
+        yield doc_batch
+
+    def convert_post_message_list_to_document(
+        self,
+        channel_object: Channel,
+        post_message_list: list[ChatMessage],
+    ) -> Document:
+        most_recent_message_datetime: datetime | None = None
+        semantic_string: str = (
+            "Channel/Post: " + channel_object.properties["displayName"]
+        )
+        post_id: str = channel_object.id
```

Comment:
Is this correct? Why is the ID the doc, the channel id? Won't this just create a single Document from the whole channel?